frontend/src/Provider/ProviderComponents/ApproveMedicationButton.jsx

```diff
@@ -0,0 +1,14 @@
+import React, { useState } from "react";
+import ApproveModal from "./ApproveModal";
+
+export default function ApproveMedicationButton({ medication }) {
+  const [toggleModal, setToggleModal] = useState(false);
+
+  return (
+    <>
+      <button className="btn btn-success m-1" onClick={() => setToggleModal(true)}>Approve</button>
+
+      <ApproveModal show={toggleModal} medication={medication} onHide={() => setToggleModal(false)}/>
+    </>
+  );

```

Comment:
Nice! This way each medication will have a custom Modal that is shown when the relevant button is clicks. 

Just to note here, the tradeoff is that you'll have way more `<ApproveModal>` components in your DOM, and they will scale with the # of buttons. 

This won't be a problem now, but such dependencies can impact bigger projects. A tradeoff to consider for modularity vs performance